ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -142,15 +226,25 @@ long getDefaultMaxOutstandingRequestBytes() {
     }
 
     static OutputType get(Env env) {
-      if (env.containsKey(OUTPUT_BUCKET)) {
+      boolean hasBigQueryOutputMode = env.containsKey(BIG_QUERY_OUTPUT_MODE);
+      if (env.containsKey(OUTPUT_BUCKET) && !hasBigQueryOutputMode) {
         return OutputType.gcs;
-      } else if (env.containsKey(OUTPUT_TOPIC)) {
+      } else if (env.containsKey(OUTPUT_TOPIC) && !hasBigQueryOutputMode) {
         return OutputType.pubsub;
-      } else if (env.containsKey(OUTPUT_TABLE)) {
-        return OutputType.bigQuery;
+      } else if (env.containsKey(OUTPUT_TABLE) || hasBigQueryOutputMode) {
+        final String outputMode = env.getString(BIG_QUERY_OUTPUT_MODE, "streaming").toLowerCase();
+        if (outputMode.equals("streaming")) {
+          return OutputType.bigQueryStreaming;
+        }
+        if (outputMode.equals("mixed")) {
+          return OutputType.bigQueryMixed;
+        }
+        if (outputMode.equals("file_loads")) {
+          return OutputType.bigQueryFiles;
+        }
+        throw new IllegalArgumentException("Unsupported BIG_QUERY_OUTPUT_MODE: " + outputMode);
       } else {
-        throw new IllegalArgumentException("Could not find at least one of [OUTPUT_BUCKET,"
-            + " OUTPUT_TOPIC, OUTPUT_TABLE] in environment");
+        return OutputType.bigQueryLoad;
```

Comment:
Why are we defaulting to bigQueryLoad now?

Reply:
`bigQueryLoad` is the one that consumes notifications from `bigQueryFiles`, and we default to it because unlike the rest of the modes it doesn't have any required output options.

Reply:
Can you add a comment to that effect?